pylib/common.py

```diff
@@ -463,6 +475,85 @@ def _init_devicestate_file():
         except FileNotFoundError:
             _init_devicestate_file()
 
+def cpu_usage(last_idle, last_total):
+    with open('/proc/stat') as f:
+        fields = [float(column) for column in f.readline().strip().split()[1:]]
+    idle, total = fields[3], sum(fields)
+    idle_delta, total_delta = idle - last_idle, total - last_total
+    last_idle, last_total = idle, total
+    utilisation = 100.0 * (1.0 - idle_delta / total_delta)
+    return (utilisation, last_idle, last_total)
```

Comment:
I also had implemented this with `psutil`, but doing it this way avoids adding that as a dependency.